src/device.ts

```diff
@@ -252,15 +252,25 @@ export class USBDevice {
         return new Promise((resolve, reject) => {
             if (!this.connected) return reject(new Error("close error: device not found"));
             if (!this.opened) return resolve();
-
-            adapter.close(this._handle)
-            .then(resolve)
-            .catch(error => {
-                reject(new Error(`close error: ${error}`));
+            const releaseInterfacePromises = [];
```

Comment:
This could read better as a map:

```typescript
const promises = this.configuration.interfaces.map(iface => this.releaseInterface(iface.interfaceNumber));
```